2d/include/pcl/2d/impl/keypoint.hpp

```diff
@@ -42,20 +42,23 @@
 #ifndef PCL_2D_KEYPOINT_HPP_
 #define PCL_2D_KEYPOINT_HPP_
 
-#include <limits>
 #include <pcl/2d/convolution.h>
 #include <pcl/2d/edge.h>
+#include <pcl/2d/keypoint.h> // for pcl::Keypoint
+
+#include <limits>
 
-//////////////////////////////////////////////////////////////////////////////
+namespace pcl {
+
+template <typename ImageType>
 void
-pcl::keypoint::harrisCorner(ImageType& output,
-                            ImageType& input,
-                            const float sigma_d,
-                            const float sigma_i,
-                            const float alpha,
-                            const float thresh)
+Keypoint<ImageType>::harrisCorner(ImageType& output,
```

Comment:
Am I missing something here? You seem to be replacing `pcl::keypoint::` with `template <class T> pcl::Keypoint<T>::`

Reply:
It wasn't handled at all, the related header might have never seen any use ever, it should not work. `ImageType` itself otherwise is nowhere to be found. This is the only explanation to how this survived. A lowercase letter instead of an upper case.